src/pages/find-offers/FindOffers.jsx

```diff
@@ -1,7 +1,52 @@
+import { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import Stack from '@mui/material/Stack'
+import { useSearchParams } from 'react-router-dom'
 import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import AppContentSwitcher from '~/components/app-content-switcher/AppContentSwitcher'
+import { styles } from '~/pages/find-offers/FindOffers.styles'
+import { student, tutor } from '~/constants'
 
 const FindOffers = () => {
-  return <PageWrapper>Find offers</PageWrapper>
+  const { userRole } = useSelector((state) => state.appMain)
+  const { t } = useTranslation()
+  const [searchParams, setSearchParams] = useSearchParams()
+  // eslint-disable-next-line no-unused-vars
+  const [view, setView] = useState(() => searchParams.get('view') || 'list')
+  const [role, setRole] = useState(() => searchParams.get('role') || userRole)
+
+  const switchOptions = {
+    left: {
+      text: t('findOffers.topMenu.tutorsOffers')
+    },
+    right: {
+      text: t('findOffers.topMenu.studentsRequests')
+    }
+  }
+
+  const changeRole = () => {
+    setRole(role === tutor ? student : tutor)
```

Comment:
Do it via prev value.
```suggestion
    setRole(prev => prev === tutor ? student : tutor)
```